src/courtroom.cpp

```diff
@@ -1870,24 +1877,34 @@ void Courtroom::handle_chatmessage(QStringList *p_contents)
   if (f_char_id >= 0)
     f_charname = ao_app->get_showname(char_list.at(f_char_id).name);
 
-  chatlogpiece *temp =
-      new chatlogpiece(f_charname, f_displayname, m_chatmessage[MESSAGE], false,
-                       m_chatmessage[TEXT_COLOR].toInt());
-  ic_chatlog_history.append(*temp);
-  if (ao_app->get_auto_logging_enabled())
-    ao_app->append_to_file(temp->get_full(), ao_app->log_filename, true);
+  if (m_chatmessage[MESSAGE] == " ") // User-created blankpost
+  {
+    m_chatmessage[MESSAGE] = ""; // Turn it into true blankpost
+  }
+  // General blankpost check
+  chatmessage_is_empty = m_chatmessage[MESSAGE] == "";
```

Comment:
Is this a _global_? What? Horrible! See if you can remove this so that the check is just `m_chatmessage[MESSAGE].isEmpty()`.

Reply:
Can do